.github/workflows/ci-core-reusable.yml

```diff
@@ -217,12 +217,28 @@ jobs:
         # We use `yarn` directly because the test launches `zk` commands in both server and EN envs.
         # An empty topmost environment helps avoid a mess when redefining env vars shared between both envs
         # (e.g., DATABASE_URL).
+        #
+        # Since `base_token` doesn't meaningfully influence the test, we use it as a flag for
+        # enabling / disabling tree during pruning.
```

Comment:
This feels _extremely_ hacky, but I guess it's better than adding even more workflows.

Reply:
As an alternative: introduce a new matrix variable, but then you'd need to specify all matrix combinations manually (at least I haven't found a way to e.g. tie one matrix variable to another). Would you like it better?